homeassistant/components/modbus/climate.py

```diff
@@ -490,6 +490,8 @@ async def _async_update(self) -> None:
                     if hvac_mode == value:
                         self._attr_hvac_mode = mode
                         break
+        else:
```

Comment:
This seems wrong, if the register is not defined, the default value is AUTO (see __init__) so it is a bit confusing to set it here.

I am pretty sure this will have some side effects.